src/sqlfluff/rules/L042.py

```diff
@@ -562,7 +561,7 @@ class SegmentCloneMap:
     """Clones a segment tree, maps from original segments to their clones."""
 
     def __init__(self, segment: BaseSegment):
-        segment_copy = copy.deepcopy(segment)
+        segment_copy = segment.copy()
```

Comment:
Ah yes -  this was causing bugs because the deep copy was copy too much. Copying the `segment` copies the `PositionMarker` which copies the `TemplatedFile`. That's too deep. The `.copy()` method on the segment manages that depth more elegantly by copying the segments but not their positions (which are immutable - so would be recreated if edited anyway).